packages/services/src/services/VideoPlayer/__tests__/VideoPlayer.test.js

```diff
@@ -0,0 +1,58 @@
+import apiDataResponse from './data/response.json';
+import root from 'window-or-global';
+import VideoPlayerAPI from '../VideoPlayer';
+
+let _scriptLoaded = false;
+
+const _partnerId = process.env.KALTURA_PARTNER_ID || 243342;
+const _uiConfId = process.env.KALTURA_UICONF_ID || 12905712;
+
+//jest.mock('../../VideoPlayer');
+
+describe('VideoPlayer', () => {
+  beforeEach(function() {
+    root.kWidget = {
+      api: {
+        doRequest: jest.fn(() => Promise.resolve(apiDataResponse)),
+      },
```

Comment:
One thing I would do, rather than setting `root.kWidget` immediately, I would actually create a method that sets it. In one of the unit tests, make a call to either api or embed, then call the method to mock `root.kWidget` afterwards. This will call the checkScript then go through the motions of loading the script.